cranelift/filetests/src/test_zkasm.rs

```diff
@@ -0,0 +1,187 @@
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use cranelift_codegen::entity::EntityRef;
+    use cranelift_codegen::ir::function::FunctionParameters;
+    use cranelift_codegen::ir::ExternalName;
+    use cranelift_codegen::isa::zkasm;
+    use cranelift_codegen::{settings, FinalizedMachReloc, FinalizedRelocTarget};
+    use cranelift_wasm::{translate_module, DummyEnvironment};
+
+    fn generate_preamble(start_func_index: usize) -> Vec<String> {
+        let mut program: Vec<String> = Vec::new();
```

Comment:
```suggestion
        // TODO: preamble should be generated by a linker and/or clift itself.
        let mut program: Vec<String> = Vec::new();
```

Reply:
I agree, I'm yet to find the exact place where this should be done.
There are https://docs.rs/cranelift-module/latest/cranelift_module/ and https://docs.rs/cranelift-object/latest/cranelift_object/ that seem to do something similar, but don't seem to be easily usable.